x-pack/plugins/security_solution/public/common/components/ml_popover/hooks/use_enable_data_feed.ts

```diff
@@ -29,9 +30,9 @@ export const useEnableDataFeed = () => {
   const enableDatafeed = useCallback(
     async (job: SecurityJob, latestTimestampMs: number, enable: boolean) => {
       submitTelemetry(job, enable);
+      setIsLoading(true);
 
       if (!job.isInstalled) {
-        setIsLoading(true);
         try {
           await setupMlJob({
```

Comment:
In theory, the UI wouldn't allow that. But `enableDatafeed` does. It would install the job and later disable it. BE could throw an exception, but I haven't tested it.

Reply:
I see after splitting we only call `setupMlJob` in the "enable", I assume the "disable" api will return an error if the job is not installed, which makes more sense to me 👍